osu.Game/Screens/OsuScreen.cs

```diff
@@ -160,7 +165,15 @@ public override void OnEntering(IScreen last)
         {
             applyArrivingDefaults(false);
 
-            backgroundStack?.Push(localBackground = CreateBackground());
+            backgroundStack?.Push(ownedBackground = CreateBackground());
+
+            background = backgroundStack?.CurrentScreen as BackgroundScreen;
+
+            if (background != ownedBackground)
+            {
+                // background may have not been replaced, at which point we don't want to track the background lifetime.
+                ownedBackground = null;
```

Comment:
I notice that the screen is not disposed here, so it looks like cleaning it up will be left to the finaliser/async disposal - was this intentional?

Reply:
I guess we can dispose here. It's not that important since at this point the background has not been loaded (and thus should be pretty low overhead).